src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -389,3 +490,359 @@ def names(self, names):
         V1ContainerImage.names = V1ContainerImage.names.setter(names)
     except Exception as ex:
         logger.debug("Error while trying to monkey patch the fix for list_node(): {}".format(str(ex)))
+
+
+def get_kubernetes_secret(api_instance, namespace, secret_name, custom_logger=None):
+    try:
+        return api_instance.read_namespaced_secret(secret_name, namespace)
+    except Exception as e:
+        handle_logging_error(custom_logger, "Error occurred when retrieving secret '{}': ".format(secret_name) + str(e))
+
+
+def handle_logging_error(custom_logger, error_string):
+    if custom_logger:
```

Comment:
Why if else here? Is there a scenario where custom_logger is not passed?

Reply:
Yes, it can None too.

I am utilising the same function based on whether custom_logger passed is or not.  Otherwise, to use the default logger to debug.
This can thus be reused later.

Example: 
https://github.com/alphaWizard/azure-cli-extensions/blob/667907ec0a1b5f6ef6c95292df836c9c5b0e8056/src/connectedk8s/azext_connectedk8s/_utils.py#L493

Reply:
should be fine